spec/system/candidate_interface/references/reference_selection/candidate_deletes_a_selected_reference_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.feature 'References' do
+  include CandidateHelper
+
+  scenario 'the candidate deletes their selected references' do
+    given_i_am_signed_in
+    and_the_reference_selection_feature_flag_is_active
+
+    when_i_receive_my_references_and_select_two_of_them
+    and_i_mark_the_select_references_section_as_complete
+    and_i_delete_one_of_the_selected_references
+    then_i_am_redirected_to_the_references_review_page
+    and_the_section_is_not_marked_as_completed
+
+    when_i_select_my_other_reference
+    and_complete_the_section
+    then_the_section_is_marked_as_completed
+
+    when_i_delete_one_of_the_selected_references
+    then_i_am_redirected_to_the_references_review_page
+    and_the_section_is_not_marked_as_completed
+
+    when_i_visit_the_select_references_page
+    then_i_am_presented_with_the_guidance
+  end
+
+  def given_i_am_signed_in
+    @candidate = create(:candidate)
+    login_as(@candidate)
+    @application = @candidate.current_application
+  end
+
+  def and_the_reference_selection_feature_flag_is_active
+    FeatureFlag.activate('reference_selection')
+  end
+
+  def when_i_receive_my_references_and_select_two_of_them
+    create_list(:reference, 3, feedback_status: :feedback_provided, application_form: @application)
+    @application.application_references[0..1].each { |reference| reference.update!(selected: true) }
+  end
+
+  def and_i_mark_the_select_references_section_as_complete
+    @application.update!(references_completed: true)
+  end
```

Comment:
Nit: worth making these two bits of manual state setup 'given' steps rather than whens.